### PR TITLE
Fix for #2748

### LINKED ISSUE #1987: Spider jimmy-johns is broken
During the global build at 2021-06-16-14-42-20, spider **jimmy-johns** failed with **0 features** and **1546 errors**.

Here's [the log](https://data.alltheplaces.xyz/runs/2021-06-16-14-42-20/logs/jimmy-johns.txt) and [the output](https://data.alltheplaces.xyz/runs/2021-06-16-14-42-20/output/jimmy-johns.geojson) ([on a map](https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/runs/2021-06-16-14-42-20/output/jimmy-johns.geojson))

### LINKED ISSUE #2748: Spider jimmy-johns is broken
During the global build at 2021-09-01-14-42-16, spider **jimmy-johns** failed with **0 features** and **1543 errors**.

Here's [the log](https://data.alltheplaces.xyz/runs/2021-09-01-14-42-16/logs/jimmy-johns.txt) and [the output](https://data.alltheplaces.xyz/runs/2021-09-01-14-42-16/output/jimmy-johns.geojson) ([on a map](https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/runs/2021-09-01-14-42-16/output/jimmy-johns.geojson))

### LINKED ISSUE #3086: Spider jimmy-johns is broken
During the global build at 2021-09-29-14-42-48, spider **jimmy-johns** failed with **0 features** and **1544 errors**.

Here's [the log](https://data.alltheplaces.xyz/runs/2021-09-29-14-42-48/logs/jimmy-johns.txt) and [the output](https://data.alltheplaces.xyz/runs/2021-09-29-14-42-48/output/jimmy-johns.geojson) ([on a map](https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/runs/2021-09-29-14-42-48/output/jimmy-johns.geojson))

### PR DESCRIPTION
**Pull request by isaacvv**
Fixes #2748
Fixes #1987
Fixes #3086

### CODE REVIEW

File: locations/spiders/jimmy_johns.py [hunk 1]
```diff
@@ -1,61 +1,53 @@
...
+    start_urls = (
+        'https://locations.jimmyjohns.com/sitemap.xml',
+        #'https://locations.jimmyjohns.com/ks/emporia/sandwiches-1298.html',
+    )
+
+    def parse(self, response):
+        
+        stores = response.xpath('//url/loc[contains(text(),"sandwiches")]/text()').extract()
+        for store in stores:
+            print(store)
```
Comment (iandees) [inline locations/spiders/jimmy_johns.py:19]: Please don't `print()` anything.

File: locations/spiders/jimmy_johns.py [hunk 2]
```diff
@@ -1,61 +1,53 @@
...
-            yield request
+        properties = {
+            'ref': response.url,
+            #'name': name,
+            'addr_full': address1[0],
+            'city': city,
+            'state': state,
+            'postcode': postcode,
+            #'country': country,
+            'phone': phone[0],
```
Comment (iandees) [inline locations/spiders/jimmy_johns.py:47]: The log (https://data.alltheplaces.xyz/ci/1308446185/jimmy_johns/log.txt) shows that some places don't extract anything for the phone number and this errors.

File: locations/spiders/jimmy_johns.py [hunk 3]
```diff
@@ -1,61 +1,53 @@
...
-                method='POST',
-                body=current_city,
-                headers=HEADERS,
-                callback=self.parse
-            )
-            yield request
+        properties = {
+            'ref': response.url,
+            #'name': name,
+            'addr_full': address1[0],
```
Comment (iandees) [inline locations/spiders/jimmy_johns.py:42]: Instead of calling `extract()` and then taking the first index of the returned array, use `extract_first()`.

**Review by iandees — CHANGES_REQUESTED** [review]

**Comment by iandees** [comment]
Also, can you please extract the `ld+json` out of the location page and use that for store attributes instead of extracting information from the HTML?

File: locations/spiders/jimmy_johns.py (hunk 1 above)
Comment (isaacvv) [reply locations/spiders/jimmy_johns.py:19]: Yeah, missed deleting that one. Thanks

**Comment by isaacvv** [comment]
> Also, can you please extract the `ld+json` out of the location page and use that for store attributes instead of extracting information from the HTML?

Can you explain this some more? I'm not sure what you mean. Thanks

**Comment by isaacvv** [comment]
> 
> 
> > Also, can you please extract the `ld+json` out of the location page and use that for store attributes instead of extracting information from the HTML?
> 
> Can you explain this some more? I'm not sure what you mean. Thanks

I think I figured this out. Thanks!

**Comment by iandees** [comment]
Great, thanks!

**Pull request merged**